packages/shorebird_cli/test/src/executables/gradlew_test.dart

```diff
@@ -329,5 +329,101 @@ Make sure you have run "flutter build apk" at least once.''',
         },
       );
     });
+
+    group('exists', () {
+      late Directory tempDir;
+      setUp(() {
+        tempDir = setUpAppTempDir();
+      });
+
+      group('when gradlew does not exist', () {
+        test('returns false', () {
+          expect(gradlew.exists(tempDir.path), isFalse);
+        });
+      });
+
+      group('when gradlew exists', () {
+        group(
+          'when on mac and linux',
+          () {
+            test('returns true', () {
+              File(
+                p.join(tempDir.path, 'android', 'gradlew'),
+              ).createSync(recursive: true);
+              expect(gradlew.exists(tempDir.path), isTrue);
+            });
+          },
+          onPlatform: {'windows': const Skip()},
+        );
+
+        group(
+          'when on windows',
+          () {
+            test('returns true', () {
+              File(
+                p.join(tempDir.path, 'android', 'gradlew.bat'),
+              ).createSync(recursive: true);
+              expect(gradlew.exists(tempDir.path), isTrue);
+            });
+          },
+          onPlatform: {
+            'mac-os': const Skip(),
+            'linux': const Skip(),
+          },
+        );
+      });
+    });
+
+    group('version', () {
+      late Directory tempDir;
+
+      setUp(() {
+        when(() => platform.isLinux).thenReturn(true);
+        when(() => platform.isMacOS).thenReturn(false);
+        when(() => platform.isWindows).thenReturn(false);
```

Comment:
This test file seems to be using a combination of `onPlatform` and mocking `platform.isOS`. We should probably stick to one or the other.